policies/recipes/steps.go

```diff
@@ -90,8 +100,335 @@ func parsePermissions(s string) (os.FileMode, error) {
+func StepArchiveExtraction(step *osconfigpb.SoftwareRecipe_Step_ArchiveExtraction, artifacts map[string]string, stepDir string) error {
+	filename, ok := artifacts[step.ArchiveExtraction.GetArtifactId()]
+	if !ok {
+		return fmt.Errorf("%q not found in artifact map", step.ArchiveExtraction.GetArtifactId())
+	}
+	switch step.ArchiveExtraction.GetType() {
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_ZIP:
+		return extractZip(filename, step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR:
+		return extractTar(filename, step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR_GZIP:
+		compressed, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer compressed.Close()
+		reader, err := gzip.NewReader(compressed)
+		if err != nil {
+			return err
+		}
+		defer reader.Close()
+		decompressed, err := ioutil.TempFile(stepDir, "archive-*.tar")
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(decompressed, reader)
+		decompressed.Close()
+		if err != nil {
+			return err
+		}
+		return extractTar(decompressed.Name(), step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR_BZIP:
+		compressed, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer compressed.Close()
+		reader := bzip2.NewReader(compressed)
+		if err != nil {
+			return err
+		}
+		decompressed, err := ioutil.TempFile(stepDir, "archive-*.tar")
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(decompressed, reader)
+		decompressed.Close()
+		if err != nil {
+			return err
+		}
+		return extractTar(decompressed.Name(), step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR_LZMA:
+		compressed, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer compressed.Close()
+		reader, err := lzma.NewReader2(compressed)
+		if err != nil {
+			return err
+		}
+		decompressed, err := ioutil.TempFile(stepDir, "archive-*.tar")
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(decompressed, reader)
+		decompressed.Close()
+		if err != nil {
+			return err
+		}
+		return extractTar(decompressed.Name(), step.ArchiveExtraction.Destination)
+	case osconfigpb.SoftwareRecipe_Step_ExtractArchive_TAR_XZ:
+		compressed, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer compressed.Close()
+		reader, err := xz.NewReader(compressed)
+		if err != nil {
+			return err
+		}
+		decompressed, err := ioutil.TempFile(stepDir, "archive-*.tar")
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(decompressed, reader)
+		decompressed.Close()
+		if err != nil {
+			return err
+		}
+		return extractTar(decompressed.Name(), step.ArchiveExtraction.Destination)
+	default:
+		return fmt.Errorf("Unrecognized archive type %q", step.ArchiveExtraction.GetType())
+	}
+}
+
+func zipIsDir(name string) bool {
+	return strings.HasSuffix(name, string(os.PathSeparator))
```

Comment:
is this right? docs for zip.FileHeader.Name say 
```
        // Name is the name of the file.
        //
        // It must be a relative path, not start with a drive letter (such as "C:"),
        // and must use forward slashes instead of back slashes. A trailing slash
        // indicates that this file is a directory and should have no data.
```

which would seem to indicate this would always be a forward slash, not a platform dependent path separator.

Reply:
You're correct, I changed it.